App/models/publication.py

```diff
@@ -8,6 +8,8 @@ class Publication(db.Model):
     title = db.Column(db.String(120), nullable=False, unique=True)
     authorId = db.Column(db.Integer, db.ForeignKey("author.id"), nullable=False)
     coauthors = db.relationship("Author", secondary=CoAuthorPublication)
+    abstract = db.Column(db.String(200), nullable=False, unique=True)
+    dateOfPublication = db.Column(db.Date, nullable=False, unique=True)
 
     def __init__(self, title, authorId, coauthors):
```

Comment:
This is the function that is called when a new Publication is created. This needs to be changed to accept an abstract and dateOfPublication and set each of them from the data it takes in.

Reply:
the controller was updated as well to accept the 2 new additions